src/System.Runtime/tests/System/Security/SecurityExceptionTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Security;
+
+using Xunit;
+
+namespace System.Tests
+{
+    public static class SecurityExceptionTests
+    {
+        private const int COR_E_SECURITY = unchecked((int)0x8013150A);
+
+        [Fact]
+        public static void Ctor_Empty()
+        {
+            var exception = new SecurityException();
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_SECURITY, validateMessage: false);
+        }
+
+        [Fact]
+        public static void Ctor_String()
+        {
+            string message = "security problem";
+            var exception = new SecurityException(message);
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_SECURITY, message: message);
+        }
+
+        [Fact]
+        public static void Ctor_String_Exception()
+        {
+            string message = "security problem";
+            var innerException = new Exception("Inner exception");
+            var exception = new SecurityException(message, innerException);
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_SECURITY, innerException: innerException, message: message);
+        }
+
+        [Fact]
+        public static void Ctor_String_Type()
+        {
+            string message = "security problem";
+            var exception = new SecurityException(message, typeof(SecurityExceptionTests));
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_SECURITY, message: message);
+            Assert.Equal(typeof(SecurityExceptionTests), exception.PermissionType);
+        }
+
+        [Fact]
+        public static void Ctor_String_Type_String()
+        {
+            string message = "security problem";
+            var exception = new SecurityException(message, typeof(SecurityExceptionTests), "permission state");
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_SECURITY, message: message);
+            Assert.Equal(typeof(SecurityExceptionTests), exception.PermissionType);
+            Assert.Equal("permission state", exception.PermissionState);
+        }
+
+        [Fact]
+        public static void Properties()
+        {
+            var exception = new SecurityException();
+
+            Assert.Null(exception.Demanded);
+            exception.Demanded = "Demanded";
```

Comment:
Looks like there are more constraints on what this can actually be on desktop, and it's failing on desktop:
```
Message :
System.InvalidCastException : Unable to cast object of type 'System.String' to type 'System.Security.IPermission'.
Stack Trace :
   at System.Security.Util.XMLUtil.SecurityObjectToXmlString(Object ob)
   at System.Security.SecurityException.set_Demanded(Object value)
   at System.Tests.SecurityExceptionTests.Properties() in D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Runtime\tests\System\Security\SecurityExceptionTests.cs:line 66
```
https://referencesource.microsoft.com/#mscorlib/system/security/util/xmlutil.cs,190